Traceback (most recent call last):
  File "/Users/morenolq/miniconda3/lib/python3.9/site-packages/jupyter_cache/executors/utils.py", line 58, in single_nb_execution
    executenb(
  File "/Users/morenolq/miniconda3/lib/python3.9/site-packages/nbclient/client.py", line 1253, in execute
    return NotebookClient(nb=nb, resources=resources, km=km, **kwargs).execute()
  File "/Users/morenolq/miniconda3/lib/python3.9/site-packages/nbclient/util.py", line 84, in wrapped
    return just_run(coro(*args, **kwargs))
  File "/Users/morenolq/miniconda3/lib/python3.9/site-packages/nbclient/util.py", line 62, in just_run
    return loop.run_until_complete(coro)
  File "/Users/morenolq/miniconda3/lib/python3.9/asyncio/base_events.py", line 647, in run_until_complete
    return future.result()
  File "/Users/morenolq/miniconda3/lib/python3.9/site-packages/nbclient/client.py", line 693, in async_execute
    await self.async_execute_cell(
  File "/Users/morenolq/miniconda3/lib/python3.9/site-packages/nbclient/client.py", line 1010, in async_execute_cell
    await self._check_raise_for_error(cell, cell_index, exec_reply)
  File "/Users/morenolq/miniconda3/lib/python3.9/site-packages/nbclient/client.py", line 904, in _check_raise_for_error
    raise CellExecutionError.from_cell_and_msg(cell, exec_reply_content)
nbclient.exceptions.CellExecutionError: An error occurred while executing the following cell:
------------------
import torch
import torchvision
import torchvision.transforms as transforms
import matplotlib.pyplot as plt
import numpy as np
import torch.nn as nn
import torch.nn.functional as F
import torch.optim as optim
from tqdm.notebook import tqdm

device = torch.device("cuda:0" if torch.cuda.is_available() else "cpu")

# Define a CNN
class Net(nn.Module):
    def __init__(self):
        super(Net, self).__init__()
        self.conv1 = nn.Conv2d(3, 6, 5)
        self.pool = nn.MaxPool2d(2, 2)
        self.conv2 = nn.Conv2d(6, 16, 5)
        self.fc1 = nn.Linear(16 * 5 * 5, 120)
        self.fc2 = nn.Linear(120, 84)
        self.fc3 = nn.Linear(84, 10)

    def forward(self, x):
        x = self.pool(F.relu(self.conv1(x)))
        x = self.pool(F.relu(self.conv2(x)))
        x = torch.flatten(x, 1) # flatten all dimensions except batch
        x = F.relu(self.fc1(x))
        x = F.relu(self.fc2(x))
        x = self.fc3(x)
        return x

# Load the data
dataset = torchvision.datasets.CIFAR10(root='./data', train=True,
                                        download=True, transform=transforms.ToTensor())

trainset, valset = torch.utils.data.random_split(dataset, [40000, 10000])

trainloader = torch.utils.data.DataLoader(trainset, batch_size=4,
                                          shuffle=True, num_workers=2)

valloader = torch.utils.data.DataLoader(valset, batch_size=4,
                                            shuffle=False, num_workers=2)

testset = torchvision.datasets.CIFAR10(root='./data', train=False,
                                        download=True, transform=transforms.ToTensor()) 
            
testloader = torch.utils.data.DataLoader(testset, batch_size=4,
                                          shuffle=False, num_workers=2)

classes = ('plane', 'car', 'bird', 'cat',
            'deer', 'dog', 'frog', 'horse', 'ship', 'truck')

# Define the model
net = Net()

# Define the loss function and optimizer
criterion = nn.CrossEntropyLoss()
optimizer = optim.SGD(net.parameters(), lr=0.001, momentum=0.9)

def train_one_epoch(net, trainloader, criterion, optimizer):
    running_loss = 0.0
    for i, data in enumerate(tqdm(trainloader)):
        # get the inputs; data is a list of [inputs, labels]
        inputs, labels = data

        # to GPU
        inputs = inputs.to(device)
        labels = labels.to(device)

        # zero the parameter gradients
        optimizer.zero_grad()

        # forward + backward + optimize
        outputs = net(inputs)
        loss = criterion(outputs, labels)
        loss.backward()
        optimizer.step()

        # print statistics
        running_loss += loss.item()
    return running_loss / len(trainloader)

def evaluate(net, valloader, criterion):
    running_loss = 0.0
    correct = 0
    total = 0
    with torch.no_grad():
        for data in tqdm(valloader):
            images, labels = data
            images = images.to(device)
            labels = labels.to(device)
            outputs = net(images)
            loss = criterion(outputs, labels)
            running_loss += loss.item()
            _, predicted = torch.max(outputs.data, 1)
            total += labels.size(0)
            correct += (predicted == labels).sum().item()
    return running_loss / len(valloader), correct / total

# Train the model
net.to(device)

for epoch in range(10):  # loop over the dataset multiple times

    train_loss = train_one_epoch(net, trainloader, criterion, optimizer)
    val_loss, val_acc = evaluate(net, valloader, criterion)
    print(f"Epoch {epoch} - Train loss: {train_loss:.3f} - Val loss: {val_loss:.3f} - Val acc: {val_acc:.3f}")

print('Finished Training')

# Evaluate the model on the test set
test_loss, test_acc = evaluate(net, testloader, criterion)
print(f"Test loss: {test_loss:.3f} - Test acc: {test_acc:.3f}")
------------------

[0;31m---------------------------------------------------------------------------[0m
[0;31mImportError[0m                               Traceback (most recent call last)
Input [0;32mIn [1][0m, in [0;36m<cell line: 1>[0;34m()[0m
[0;32m----> 1[0m [38;5;28;01mimport[39;00m [38;5;21;01mtorch[39;00m
[1;32m      2[0m [38;5;28;01mimport[39;00m [38;5;21;01mtorchvision[39;00m
[1;32m      3[0m [38;5;28;01mimport[39;00m [38;5;21;01mtorchvision[39;00m[38;5;21;01m.[39;00m[38;5;21;01mtransforms[39;00m [38;5;28;01mas[39;00m [38;5;21;01mtransforms[39;00m

File [0;32m~/miniconda3/lib/python3.9/site-packages/torch/__init__.py:197[0m, in [0;36m<module>[0;34m[0m
[1;32m    195[0m     [38;5;28;01mif[39;00m USE_GLOBAL_DEPS:
[1;32m    196[0m         _load_global_deps()
[0;32m--> 197[0m     [38;5;28;01mfrom[39;00m [38;5;21;01mtorch[39;00m[38;5;21;01m.[39;00m[38;5;21;01m_C[39;00m [38;5;28;01mimport[39;00m [38;5;241m*[39m  [38;5;66;03m# noqa: F403[39;00m
[1;32m    199[0m [38;5;66;03m# Appease the type checker; ordinarily this binding is inserted by the[39;00m
[1;32m    200[0m [38;5;66;03m# torch._C module initialization code in C[39;00m
[1;32m    201[0m [38;5;28;01mif[39;00m TYPE_CHECKING:

[0;31mImportError[0m: dlopen(/Users/morenolq/miniconda3/lib/python3.9/site-packages/torch/_C.cpython-39-darwin.so, 0x0002): Library not loaded: @rpath/libtorch_cpu.dylib
  Referenced from: <BA93ABA0-52DE-3A68-8181-9B1B82C6F0AF> /Users/morenolq/miniconda3/lib/python3.9/site-packages/torch/lib/libtorch_python.dylib
  Reason: tried: '/Users/morenolq/miniconda3/lib/python3.9/site-packages/torch/lib/libtorch_cpu.dylib' (no such file), '/Users/distiller/project/miniconda/envs/wheel_py39/lib/libtorch_cpu.dylib' (no such file), '/System/Volumes/Preboot/Cryptexes/OS/Users/distiller/project/miniconda/envs/wheel_py39/lib/libtorch_cpu.dylib' (no such file), '/Users/distiller/project/miniconda/envs/wheel_py39/lib/libtorch_cpu.dylib' (no such file), '/System/Volumes/Preboot/Cryptexes/OS/Users/distiller/project/miniconda/envs/wheel_py39/lib/libtorch_cpu.dylib' (no such file), '/Users/distiller/project/miniconda/envs/wheel_py39/lib/libtorch_cpu.dylib' (no such file), '/System/Volumes/Preboot/Cryptexes/OS/Users/distiller/project/miniconda/envs/wheel_py39/lib/libtorch_cpu.dylib' (no such file), '/Users/morenolq/miniconda3/lib/python3.9/site-packages/torch/lib/libtorch_cpu.dylib' (no such file), '/Users/distiller/project/miniconda/envs/wheel_py39/lib/libtorch_cpu.dylib' (no such file), '/System/Volumes/Preboot/Cryptexes/OS/Users/distiller/project/miniconda/envs/wheel_py39/lib/libtorch_cpu.dylib' (no such file), '/Users/distiller/project/miniconda/envs/wheel_py39/lib/libtorch_cpu.dylib' (no such file), '/System/Volumes/Preboot/Cryptexes/OS/Users/distiller/project/miniconda/envs/wheel_py39/lib/libtorch_cpu.dylib' (no such file), '/Users/morenolq/miniconda3/lib/python3.9/site-packages/torch/lib/libtorch_cpu.dylib' (no such file), '/Users/morenolq/miniconda3/bin/../lib/libtorch_cpu.dylib' (no such file), '/Users/morenolq/miniconda3/bin/../lib/libtorch_cpu.dylib' (no such file), '/usr/local/lib/libtorch_cpu.dylib' (no such file), '/usr/lib/libtorch_cpu.dylib' (no such file, not in dyld cache)
ImportError: dlopen(/Users/morenolq/miniconda3/lib/python3.9/site-packages/torch/_C.cpython-39-darwin.so, 0x0002): Library not loaded: @rpath/libtorch_cpu.dylib
  Referenced from: <BA93ABA0-52DE-3A68-8181-9B1B82C6F0AF> /Users/morenolq/miniconda3/lib/python3.9/site-packages/torch/lib/libtorch_python.dylib
  Reason: tried: '/Users/morenolq/miniconda3/lib/python3.9/site-packages/torch/lib/libtorch_cpu.dylib' (no such file), '/Users/distiller/project/miniconda/envs/wheel_py39/lib/libtorch_cpu.dylib' (no such file), '/System/Volumes/Preboot/Cryptexes/OS/Users/distiller/project/miniconda/envs/wheel_py39/lib/libtorch_cpu.dylib' (no such file), '/Users/distiller/project/miniconda/envs/wheel_py39/lib/libtorch_cpu.dylib' (no such file), '/System/Volumes/Preboot/Cryptexes/OS/Users/distiller/project/miniconda/envs/wheel_py39/lib/libtorch_cpu.dylib' (no such file), '/Users/distiller/project/miniconda/envs/wheel_py39/lib/libtorch_cpu.dylib' (no such file), '/System/Volumes/Preboot/Cryptexes/OS/Users/distiller/project/miniconda/envs/wheel_py39/lib/libtorch_cpu.dylib' (no such file), '/Users/morenolq/miniconda3/lib/python3.9/site-packages/torch/lib/libtorch_cpu.dylib' (no such file), '/Users/distiller/project/miniconda/envs/wheel_py39/lib/libtorch_cpu.dylib' (no such file), '/System/Volumes/Preboot/Cryptexes/OS/Users/distiller/project/miniconda/envs/wheel_py39/lib/libtorch_cpu.dylib' (no such file), '/Users/distiller/project/miniconda/envs/wheel_py39/lib/libtorch_cpu.dylib' (no such file), '/System/Volumes/Preboot/Cryptexes/OS/Users/distiller/project/miniconda/envs/wheel_py39/lib/libtorch_cpu.dylib' (no such file), '/Users/morenolq/miniconda3/lib/python3.9/site-packages/torch/lib/libtorch_cpu.dylib' (no such file), '/Users/morenolq/miniconda3/bin/../lib/libtorch_cpu.dylib' (no such file), '/Users/morenolq/miniconda3/bin/../lib/libtorch_cpu.dylib' (no such file), '/usr/local/lib/libtorch_cpu.dylib' (no such file), '/usr/lib/libtorch_cpu.dylib' (no such file, not in dyld cache)

